diff --git a/entity.module b/entity.module
--- a/entity.module
+++ b/entity.module
@@ -877,9 +877,11 @@ function _entity_defaults_rebuild($entity_type) {
     $entities = array();
     foreach (module_implements($hook) as $module) {
       foreach ((array) module_invoke($module, $hook) as $name => $entity) {
-        $entity->{$keys['name']} = $name;
-        $entity->{$keys['module']} = $module;
-        $entities[$name] = $entity;
+        if ($entity) {
+          $entity->{$keys['name']} = $name;
+          $entity->{$keys['module']} = $module;
+          $entities[$name] = $entity;
+        }
       }
     }
     drupal_alter($hook, $entities);
